src/pages/products-page/ProductsPage.jsx

```diff
@@ -1,7 +1,27 @@
 import { Navbar, Sidebar, ProductsCard } from "../../components/index";
+import { useState, useEffect } from "react";
+import axios from "axios"; 
 import "./products-page.css";
 
 const ProductsPage = () => {
+     
+    useEffect(()=>{
+        fetchProducts()
+    }, []);
+
+    const [loader, setLoader] = useState(false);
+    const [products, setProducts] = useState([]);
```

Comment:
Just a suggestion - when you create states just create them on top of everything inside functional component.

Reply:
done